[AST] divination update, checklist reorder, random cleanup

otocephaly opened the pull request:
Hiiiii

I updated display order on some core modules in order to be able to rearrange things in the checklist
Divination was updated based on what's listed in the balance

This is what both look like after running it using example log below
![image](https://user-images.githubusercontent.com/89435408/199157368-4fe95fac-8528-4031-a637-9e192a1c3d2b.png)

![image](https://user-images.githubusercontent.com/89435408/199157404-46582a5b-30f2-43ab-b4d0-9ffc54b2c700.png)

https://www.fflogs.com/reports/a:KtJ3NaWZm48kDLXF#fight=4&source=6

otocephaly committed:
[AST] display order checklist

adding display order to checklist items

otocephaly committed:
[AST] display order checklist

updating display order for the checklist

otocephaly committed:
[AST] collective and order

updating display order on checklist and adding CU to checklist

otocephaly committed:
[AST] removing period

removed period to be consistent with other checklist titles

otocephaly committed:
[AST] kill lightspeed and update divination

killed lightspeed
updated divination to include standard rotation items

otocephaly committed:
[AST] change log update

update changelog based on updates

otocephaly committed:
yarn extract

yarn extract

otocephaly committed:
Merge branch 'xivanalysis:endwalker' into random_ast_cleanup

eatmorehufu committed:
Migrate positionals types and definitions from core/jobs to data to better support layering. (#1684)

* Add potency modelling to data/ACTIONS from the positional module

This will make layering a little easier as patches change potencies.

* Add changelog for NIN which also changed.

* Actually use a correct changelog date.

* Dates are too hard

* Fix comments

* Migrate MNK potencies to the data model.

---------

Co-authored-by: Noah Engelberth <26804378@users.noreply.github.com>

otocephaly committed:
Merge branch 'xivanalysis:endwalker' into random_ast_cleanup

otocephaly committed:
Revert "Migrate positionals types and definitions from core/jobs to data to better support layering. (#1684)"

This reverts commit 28c7def87bc4974eddef061445d65637b0db6b39.

otocephaly committed:
Merge branch 'endwalker' into random_ast_cleanup

otocephaly committed:
updated changelog date

updated date since it's been a very long time since this PR was originally submitted

otocephaly committed:
updated for may 2023

updated PR since time has passed since original PR and this needed to be updated

otocephaly committed:
removing defensive portion in oGCDs

see title

otocephaly committed:
adding macrocosmos to defensives

in the spirit of this module, macrocosmos can be used for weaving. adding this in will allow for better planning for the AST

otocephaly commented:
Hiii

Subsequent to the many changes in the meantime, I've updated the relevant sections which included removing the defensives from the checklist. In spirit of this and since it'll be handy for planning purposes to have them on this list, I've added collective unconsciousness (as discussed), macrocosmos since it's relatively under utilized without planning, and earthly star since we don't have a specific section dealing with it. Here are the updated screenshots as a result

Please note the following however,
Bump to 6.4 will come at a later PR to separate that out and keep that PR simple
Undraw seems to come up in the arcana logs whenever there is a redraw. Will work on fixing that (or someone can help if they wish)

cc: @suShirow 

Divination:
![image](https://github.com/xivanalysis/xivanalysis/assets/89435408/f2e969a2-579e-415f-b3d1-914f43e8fc93)

Defensives:
![image](https://github.com/xivanalysis/xivanalysis/assets/89435408/12db3d96-50ad-443c-937a-84d1f2bc3d2a)

Checklist after reordering:
![image](https://github.com/xivanalysis/xivanalysis/assets/89435408/6137ef3a-61f4-4bbc-805d-4e325fbcacfc)

AzariahAeroborn commented on src/parser/jobs/ast/modules/Lightspeed.tsx:
```diff
@@ -1,199 +0,0 @@
-					.type('action'),
-				this.onCast,
-			)
-			this.castTimeIndex = this.castTime.setTimeAdjustment('all', LIGHTSPEED_REDUCTION)
```
This code (line 76, and the matching code on lines 94 and 95 for tryCloseWindow) is critical for marking spells as instant casts during a Lightspeed window, and needs to be ported to _somewhere_.  There's a visibility/maintainability argument for keeping a separate Lightspeed module to do this castTime modification, or I think having the Lightspeed handling part of the Divination module handle it would be fine as well (though my preference is the more discoverable separate Lightspeed module, even though it wouldn't be suggesting / recommending anything).

Without this code, we're dinging multi-weaves in lightspeed (see http://localhost:3000/fflogs/A6TWmwzX9xkQbtva/117/84)

otocephaly replied:
very good point! I see what you mean
I'll add in a lightspeed module and reference it in divination as suggested

otocephaly replied:
added lightspeed section
lemme know if it's sufficient
thank you ♥

AzariahAeroborn reviewed (changes requested):
The MAJOR suggestion for a single usage of Lightspeed that was decoupled from Divination seems a little stringent to me, but I did see you and Sushi had discussed this previously, so if you're comfortable with that level then we can roll with it.  See comment on the castTime adjustment from Lightspeed below.

otocephaly commented:
sushi didn't state a level at which to put it, but I think major makes sense based on her statement
I'll leave it at major for now and see if there are any user or balance feedback

(here's the conversation for my own reference in case I forget in the future)
![image](https://github.com/xivanalysis/xivanalysis/assets/89435408/3460fca7-ab0e-4aa2-9472-6c03021d50ee)

otocephaly committed:
adding back lightspeed for weaving

added back lightspeed for weaving purposes (castTime) although no suggestions will come out from this

otocephaly committed:
no title handle lightspeed

I do not need these and it saves me from a yarn extract which failed in circle ci oop

otocephaly committed:
I guess I did need it

handle added. I avoided adding the other thing because I don't want to yarn extract

AzariahAeroborn commented on src/parser/jobs/ast/modules/Lightspeed.tsx:
```diff
@@ -0,0 +1,44 @@
+	@dependency private data!: Data
+	@dependency private castTime!: CastTime
+
+	private castedTime: number | undefined = undefined //when lightspeed was casted
```
@otocephaly 
Might be worth cleaning this up in a follow-up, since this is really going unused.  I realize it was probably lifted from the previous incarnation of the code and so I'm not going to block on it, but might be worth removing the vestigal code so future maintainers don't have to wonder what's going on here.

otocephaly replied:
oop I thought I cleaned it up
I'll keep in mind for a clean up in the near future then
thanks for the advice

suShirow commented:
I'm late to the party - apologies that my availability this patch has been tight. 
It appears to be the general consensus that 2min burst with lightspeed feels awful and defeats the purpose, since you'll end up drifting cards out of burst. 

There's some specific situations where it might be useful to decouple it 
This expac, off the top of my head, I can only think of holding it for prog situations for raise and maybe P3S fountain of fire, but even that you can get away with not using LS. 

So I concur with leaving it as major
Appreciate ya'll handling all of this

suShirow commented:
I realized Aza's point was "a single usage", 
Maybe we can have it as 
1 decouple: medium
2 decouples: major

But we can re-evaluate that
